game-core/src/main/java/org/triplea/game/server/HeadlessServerSetupPanelModel.java

```diff
@@ -21,7 +39,64 @@
 
   @Override
   public void showSelectType() {
-    new ServerModel(gameSelectorModel, this, null).createServerMessenger();
+    new ServerModel(gameSelectorModel, this, null, new LaunchAction() {
+      @Override
```

Comment:
Large anonymous inner class here, perhaps it's better this be a standalone class? I notice as well the headed version is its own class, yet this one is defined as an anonymous inner. Perhaps it would be more consistent as a stand-alone class so then we have the headed and headless versions.

Reply:
Yeah originally both implementations were anonymous classes, then I noticed I actually needed the headed version more than once, so I had to create a separate class.

Reply:
Will you convert this to a stand-alone class? The lack of symmetry is bothersome not to mention the anonymous inner class length. AFAIK anonymous inner classes are best short, not for full blown implementations.

Reply:
Yeah, I will